governance-chat/program/src/state.rs

```diff
@@ -0,0 +1,94 @@
+//! Program state
+
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use solana_program::{
+    account_info::AccountInfo, clock::UnixTimestamp, program_error::ProgramError, pubkey::Pubkey,
+};
+use spl_governance::tools::account::{assert_is_valid_account, AccountMaxSize};
+
+/// Defines all GovernanceChat accounts types
+#[repr(C)]
+#[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
+pub enum GovernanceChatAccountType {
+    /// Default uninitialized account state
+    Uninitialized,
+
+    /// Chat message
+    ChatMessage,
+}
+
+/// Chat message body
+#[derive(Clone, Debug, PartialEq, BorshDeserialize, BorshSerialize, BorshSchema)]
+pub enum MessageBody {
+    /// Text message encoded as utf-8 string
+    Text(String),
+
+    /// Emoticon encoded using utf-8 characters
+    Reaction(String),
```

Comment:
nit: Is this meant to explain how to display this on the frontend as not a message, but just a reaction under the message? If so, can you explain that in the comment above?

Reply:
Done